sqlite3_opt_serialize.go

```diff
@@ -0,0 +1,84 @@
+// +build !libsqlite3 sqlite_serialize
+
+package sqlite3
+
+/*
+#ifndef USE_LIBSQLITE3
+#include <sqlite3-binding.h>
+#else
+#include <sqlite3.h>
+#endif
+#include <stdlib.h>
+#include <stdint.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"unsafe"
+)
+
+// Serialize returns a byte slice that is a serialization of the database.
+//
+// See https://www.sqlite.org/c3ref/serialize.html
+func (c *SQLiteConn) Serialize(schema string) ([]byte, error) {
+	if schema == "" {
+		schema = "main"
+	}
+	var zSchema *C.char
+	zSchema = C.CString(schema)
+	defer C.free(unsafe.Pointer(zSchema))
+
+	var sz C.sqlite3_int64
+	ptr := C.sqlite3_serialize(c.db, zSchema, &sz, 0)
+	if ptr == nil {
+		return nil, fmt.Errorf("serialize failed")
+	}
+	defer C.sqlite3_free(unsafe.Pointer(ptr))
+
+	if sz > C.sqlite3_int64(math.MaxInt) {
```

Comment:
@mattn FYI this constant was introduced in Go 1.17. The GitHub actions suggest that is the minimum required version for this library, but the go.mod file still says 1.16. Just wanted to confirm you have no concerns about this.

Reply:
Is this constant an operating system cpu architecture dependent?

Reply:
The value is `GOARCH`-dependent, but it is always defined (in 1.17+).

Reply:
@mattn has approved this PR, so I'll assume this conversation is resolved.